pkg/apis/management.cattle.io/v3/machine_types.go

```diff
@@ -300,34 +323,47 @@ var (
 	NodeDriverConditionInactive   condition.Cond = "Inactive"
 )
 
-type Condition struct {
-	// Type of cluster condition.
-	Type string `json:"type"`
-	// Status of the condition, one of True, False, Unknown.
-	Status v1.ConditionStatus `json:"status"`
-	// The last time this condition was updated.
-	LastUpdateTime string `json:"lastUpdateTime,omitempty"`
-	// Last time the condition transitioned from one status to another.
-	LastTransitionTime string `json:"lastTransitionTime,omitempty"`
-	// The reason for the condition's last transition.
-	Reason string `json:"reason,omitempty"`
-	// Human-readable message indicating details about last transition
-	Message string `json:"message,omitempty"`
-}
-
 type NodeDriverSpec struct {
-	DisplayName string `json:"displayName"`
-	Description string `json:"description"`
-	URL         string `json:"url" norman:"required"`
-	ExternalID  string `json:"externalId"`
-	Builtin     bool   `json:"builtin"`
-	Active      bool   `json:"active"`
-	// If AddCloudCredential is true, then the cloud credential schema is created
-	// regardless of whether the node driver is active.
-	AddCloudCredential bool     `json:"addCloudCredential"`
-	Checksum           string   `json:"checksum"`
-	UIURL              string   `json:"uiUrl"`
-	WhitelistDomains   []string `json:"whitelistDomains,omitempty"`
+	// DisplayName specifies the publicly visible name of the driver shown in the Rancher UI.
+	// +kubebuilder:validation:MaxLength=57
```

Comment:
Just curious: why is the max length 57?

Reply:
https://github.com/rancher/rancher/pull/50644#discussion_r2162363141